appcues/src/main/java/com/appcues/ui/AppcuesActivity.kt

```diff
@@ -40,6 +41,24 @@ internal class AppcuesActivity : AppCompatActivity() {
 
     private val logcues: Logcues by lazy { scope.get() }
 
+    private val onBackPressCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() {
+            // disable our own back press callback
+            this.isEnabled = false
+
+            // then check if we have any other callback, i.e. debugger
+            if (onBackPressedDispatcher.hasEnabledCallbacks()) {
+                // call it if so
+                onBackPressedDispatcher.onBackPressed()
+                // then re-enable our own callback
+                this.isEnabled = true
+            } else {
+                // otherwise initiate dismissal
+                viewModel.onBackPressed()
+            }
+        }
+    }
```

Comment:
I believe this is functionally equivalent to what we had before, but this is the one I'd appreciate an extra careful look on please. I tested and when i have the debugger open on top of an experience, it appears to work the same as before: first back collapses the debugger, then second back dismisses the experience.

Reply:
it looks  similar, I'm interested to know if there was any specific scenario that required you to change the state of "isEnabled" to false and back to enabled, instead of just disabling on the else branch

Reply:
I believe if we do not set `isEnabled` to false, then the check on line 50 will always return true
```
if (onBackPressedDispatcher.hasEnabledCallbacks()) {
```
and we'll enter into an infinite loop calling our own back press handler.

Reply:
oh that makes sense, we need to account for this callback when asking if there is any enabled callback

Reply:
I'll add a comment in the code to make it clear why it is doing that!